File: truth_set/python/logic/compute/calculateTotalCompensation.py
def calculateTotalCompensation(employee):
    bonus = employee.baseSalary * employee.bonusPercentage
    tax = bonus * employee.taxPercentage
    return employee.baseSalary + bonus - tax

# Usage
class Employee:
    def __init__(self):
        self.baseSalary = 50000
        self.bonusPercentage = 0.10  # 10%
        self.taxPercentage = 0.20    # 20%

employee = Employee()
print(calculateTotalCompensation(employee))
\ No newline at end of file

File: truth_set/javascript/logic/compute/calculateTotalCompensation.js
function calculateTotalCompensation(employee) {
    var bonus = employee.baseSalary * employee.bonusPercentage;
    var tax = bonus * employee.taxPercentage;
    return employee.baseSalary + bonus - tax;
}
// Usage
var employee = {
    baseSalary: 50000,
    bonusPercentage: 0.10, // 10%
    taxPercentage: 0.20 // 20%
};
console.log(calculateTotalCompensation(employee));